docs/ANONYMITY_NETWORKS.md

```diff
@@ -1,149 +1,176 @@
+
+To connect your local node to the AN, you need a running instance for a socket to enter.
+```
+apt info tor # apt install
+apt info i2p
+```
 
-### Outbound Connections
 
-Connecting to an anonymous address requires the command line option
-`--tx-proxy` which tells `monerod` the ip/port of a socks proxy provided by a
-separate process. On most systems the configuration will look like:
+## Outbound connections
 
+### Proxy node traffic
+
+The option `--proxy` allows for outgoing connections to go over a local socket as an entry point into the AN
+```
+./monerod --proxy 127.0.0.1:5050 #tor
+./monerod --proxy 127.0.0.1:7657 #i2p
+``` 
+The `monerod` will use the proxy _and_ clearnet, unless you bind the p2p to localhost. You can close clearnet by binding p2p ip `0.0.0.0` to localhost.
 ```
---tx-proxy tor,127.0.0.1:9050,10
---tx-proxy i2p,127.0.0.1:9000
+# monero.conf 
+proxy 127.0.0.1:5050    # tor out(going)
+p2p-bind-ip 127.0.0.1   # no p2p outside ANs; default is INADDR_ANY
 ```
 
-which tells `monerod` that ".onion" p2p addresses can be forwarded to a socks
-proxy at IP 127.0.0.1 port 9050 with a max of 10 outgoing connections and
-".b32.i2p" p2p addresses can be forwarded to a socks proxy at IP 127.0.0.1 port
-9000 with the default max outgoing connections.
+The option `--proxy` will also redirect DNS over AN, unless you set `--proxy-allow-dns-leaks` to use the default dns resolver, see [#7616 (comment)](https://github.com/monero-project/monero/pull/7616#issuecomment-892736115).
 
-If desired, peers can be manually specified:
 
+### Connect to hidden service
+
+Hidden services are used for relaying transactions not blocks, be it from peers or local RPC. This adds the benefit of reaching a peer without revealing tx into clearnet (no exit relay).
+
+Add option `--tx-proxy` to connect to a hidden service,
 ```
---add-exclusive-node rveahdfho7wo4b2m.onion:28083
---add-peer rveahdfho7wo4b2m.onion:28083
+proxy 127.0.0.1:5050           # tor in & out
+tx-proxy=tor,127.0.0.1:9050,16 # onions out; [,disable_noise]
+tx-proxy=i2p,127.0.0.1:9000    # i2p
 ```
+which tells `monerod` that `.onion` p2p addresses can be forwarded to a socks
+proxy at IP 127.0.0.1 port 9050 with a max of 16 outgoing connections and
+".b32.i2p" p2p addresses can be forwarded to a socks proxy at IP 127.0.0.1 port
+9000 with the default max outgoing connections. The local tx is now exclusively sent over onions and withholds until an onion peer is found.
 
-Either option can be listed multiple times, and can specify any mix of Tor,
-I2P, and IPv4 addresses. Using `--add-exclusive-node` will prevent the usage of
-seed nodes on ALL networks, which will typically be undesirable.
+Adding `disable_noise` will tell the peer to relay the tx over tor instead of onions. It disables Dandelion++ and makes network propagation faster, but degrades anonymity, because the tx will propagate outside the AN. 
```

Comment:
The information about `disable_noise` is incorrect. By default a dummy message is sent every 10-15 seconds (randomized) to make timing correlation attacks over tor/i2p more difficult. This uses more bandwidth and causes its own "noise" (signature) over the network - adversaries will know that you are running monero but it should be harder to trace transactions. The situation is:

 * If an outgoing I2P connection is available, this is always preferred for transactions
 * If an outgoing Tor connection is available with no I2P outgoing connections, then Tor will be used for transactions
 * The daemon will refuse to send transactions if any AN is enabled but has no outgoing connections
 * Dandelion++ is not used over the AN, but the remote service receiving the tx over an AN is expected to use it after a randomized delay (no way to detect or enforce this policy however)
 * `disable_noise` only affects whether dummy messages are sent, but does not change any of the above rules.

Reply:
@selsta reminded me that one of the points is incorrect. A delay from i2p/tor->clearnet is only done when noise is enabled, there is no benefit to the delay when `disable_noise` is used.

Reply:
Wrong again, thanks @selsta .This section still needs to be updated because the language isn't quite correct. I updated my bullet-points from the first comment to be more accurate.

The most glaring issue is `tx will propagate outside the AN` is not accurate. When any AN is enabled, a local transaction is either sent over any AN or not sent at all. The `disable_noise` policy disables:
 * Periodic sending of dummy messages
 * Randomized delay on the hidden service side
 * Dandelion++ sending on the hidden service side (immediate flood/fluff)